src/main.cpp

```diff
@@ -3956,9 +3961,12 @@ bool static LoadBlockIndexDB()
     {
         return true;
     }
-    chainActive.SetTip(it->second);
 
-    PruneBlockIndexCandidates();
+    {
+        LOCK(cs_main);
+        chainActive.SetTip(it->second);
+        PruneBlockIndexCandidates();
```

Comment:
I don't think you need to lock cs_main here, it looks like it's already locked ... there is an assertlockheld() at the beginning of this function.

Reply:
thanks, clang is actually not smart enough to realize what that means, so gives the warning.  I will remove the lock here.